prov/sm2/src/sm2.h

```diff
@@ -279,13 +282,11 @@ struct sm2_ep {
 	size_t rx_size;
 	const char *name;
 	uint64_t msg_id;
-	struct sm2_region *volatile region;
-	// if double locking is needed, shm region lock must
-	// be acquired before any shm EP locks
+	int self_fiaddr;
```

Comment:
The global ids are not the same as the fiaddrs. Rename to something that indicates this is part of the internal naming index, not the fiaddr. In shm we use "shm_id" which you can use as well unless you have something else in mind

Reply:
renamed to gid